### PR TITLE
Adds Banner support for UnityAds

### PR DESCRIPTION
**Pull request by RossRothenstine**
## Description

This PR will add UnityAds Banner support and does some small cleanup to the codebase. Note that UnityAds Banner currently only supports a single banner instance.

## Testing

- iPhone XS 12.0
- iPhone 8+ 11.0

### CODE REVIEW

**Comment by chauduyphanvu1** [comment]
@RossRothenstine, I'm getting conflicting logs in my testing:

```
2018-11-12 17:06:30.598035-0500 Canary[91171:3855170] MOPUB: Loading ad with MoPub server URL: https://ads.mopub.com/m/ad
{"abt":"{\"adcolony\":{\"token\":\"1\"},\"facebook\":{\"token\":\"eJxFUNtq4zAQ\\\/Rc9y8ayZMkO9EGXcSPiSF7bSQnLIpKQ0tJNE\\\/bGlmX\\\/fSWnSx9n5lzmnD\\\/IwfTgh1WYdj2gBcFohHG03gVr0AK1RpmSFjSraqYyxkmZ1ZxApnjJpBEVbZRGGHVeyy7S0ek1bEb0ITLZdVqTkuakoZwXjWBVUfII8eMWhnE+5iTOg\\\/cTmPkFa1oZD0aDBFXrDExbZ0zUTSap4ZkBQSum21Y3SWgdlHQOhrCCXWTtyuZHdya\\\/jsvtc\\\/fing7n9u306e4ufWVWQcteKtvZKWE\\\/U8xwhQVuMCkwKTHhmIgvSRSMlVOKMMKwtTqlSAp6AHBhCfZ+OaFFHSlI9r3a2C7VVeUsLyJMbZyZ+zhezvn5cv15yPX+df\\\/tLd42bvYuIrHtI+QgyCMpP7QfrJmWaEFFlTqKgGefGo0u7339N1nLVbKQ1+vXUxy1HyDcJAXl1aneH+PWgBtvWemtgCR4+Z7o3kAXp981D5y9t5MsYujZhUSXv\\\/8AEviHog==\"},\"tapjoy\":{\"token\":\"1\"},\"applovin_sdk\":{\"token\":\"\"}}","av":"5.4.0","dn":"x86_64","force_gdpr_applies":"0","vv":"3","ats":"1","gdpr_applies":"0","h":"2436","udid":"ifa:DCEAEB8C-EDF8-4789-A3D6-DE7354CFFC96","nv":"5.4.0","ct":"2","w":"1125","v":"8","id":"a52f53a2c5cb4c36a14a4f5fca3b7182","mr":"1","bundle":"com.mopub.Canary","z":"-0500","sc":"3.0","current_consent_status":"unknown","o":"p"}
2018-11-12 17:06:30.655288-0500 Canary[91171:3855170] MOPUB: There are 1 ad responses
2018-11-12 17:06:30.655504-0500 Canary[91171:3855170] MOPUB: Banner ad view is fetching ad network type: custom
2018-11-12 17:06:30.655926-0500 Canary[91171:3855170] MOPUB: Looking for custom event class named UnityAdsBannerCustomEvent.
2018-11-12 17:06:30.656123-0500 Canary[91171:3855170] I/UnityAds: +[UnityAdsBanner destroy] (line:34) :: Unity Ads destroying current banner ad unit.
2018-11-12 17:06:30.658683-0500 Canary[91171:3855170] I/UnityAds: +[UnityAdsBanner loadBanner:] (line:16) :: Unity Ads opening banner ad unit for placement bannerads
2018-11-12 17:06:31.191096-0500 Canary[91171:3855170] W/UnityAds: +[USRVApiSdk WebViewExposed_logWarning:callback:] (line:53) :: Could not show banner due to No fill for placement bannerads
```

Why is the Unity Ads SDK saying it's "opening" the banner and then declaring there's no fill for that ad unit? Also we are not seeing consistent fill in order to certify banner - does the team have any test Unity Ads placements that fill 100% of the time?

**Comment by RossRothenstine** [comment]
Will need to update that log statement. We log that when they attempt to load the banner, but load =/= show.

File: UnityAds/MPUnityRouter.m [hunk 1]
```diff
@@ -51,12 +58,12 @@ - (void)initializeWithGameId:(NSString *)gameId
...
         [mediationMetaData setVersion:[[MoPub sharedInstance] version]];
         [mediationMetaData commit];
+        [UnityAdsBanner setDelegate:self];
         [UnityAds initialize:gameId delegate:self];
     });
 }
 
-- (void)requestVideoAdWithGameId:(NSString *)gameId placementId:(NSString *)placementId delegate:(id<MPUnityRouterDelegate>)delegate;
-{
+-(void)updateConsentStatus {
```
Comment (rsatrasala) [inline UnityAds/MPUnityRouter.m]: Lets clean this up to use `canCollectPersonalInfo` to be consistent with others.

File: UnityAds/UnityAdsBannerCustomEvent.m [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    [UnityAdsBanner destroy];
+}
+
+-(void)requestAdWithSize:(CGSize)size customEventInfo:(NSDictionary *)info {
+    NSString *gameId = info[kMPUnityRewardedVideoGameId];
+    self.placementId = info[kUnityAdsOptionPlacementIdKey];
+    if (self.placementId == nil) {
+        self.placementId = info[kUnityAdsOptionZoneIdKey];
+    }
+    if (self.placementId == nil) {
```
Comment (rsatrasala) [inline UnityAds/UnityAdsBannerCustomEvent.m]: What about nil checks for gameId?

File: UnityAds/MPUnityRouter.m (hunk 1 above)
Comment (chauduyphanvu1) [reply UnityAds/MPUnityRouter.m]: LMK if we also want to remove the outer `[[MoPub sharedInstance] currentConsentStatus] == MPConsentStatusConsented || [[MoPub sharedInstance] currentConsentStatus] == MPConsentStatusDenied` check, too.

Comment (rsatrasala) [reply UnityAds/MPUnityRouter.m]: Yes. The only condition is related to `gdprApplies`. I will inform them over the email about this change.

**Pull request merged**